src/features/Authentication/organisms/SignUpForm/SignUpForm.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { Formik, Form, Field } from 'formik';
+import * as Yup from "yup";
+import { TextField } from 'formik-mui';
+import Box from '@mui/material/Box';
+import Grid from '@mui/material/Grid';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import CircularProgress from '@mui/material/CircularProgress';
+import { useMutation } from 'urql';
+import Alert from '@mui/material/Alert';
+import AlertTitle from '@mui/material/AlertTitle';
+import { CREATE_USER } from 'mutations/CreateUser/createUser';
+import { AUTH_VALIDATION_SETS } from 'constants/auth';
+
+
+const SIGNUP_SCHEMA = Yup.object().shape({
+  firstName: Yup.string()
+    .min(AUTH_VALIDATION_SETS.MIN_LENGTH, `First name can't be less then ${AUTH_VALIDATION_SETS.MIN_LENGTH} letters`)
+    .max(AUTH_VALIDATION_SETS.MAX_LENGTH, `First name should be less then ${AUTH_VALIDATION_SETS.MIN_LENGTH} letters`)
+    .required('Required'),
+  lastName: Yup.string()
+    .min(AUTH_VALIDATION_SETS.MIN_LENGTH, `First name can't be less then ${AUTH_VALIDATION_SETS.MIN_LENGTH} letters`)
+    .max(AUTH_VALIDATION_SETS.MAX_LENGTH, `First name should be less then ${AUTH_VALIDATION_SETS.MIN_LENGTH} letters`)
+    .required('Required'),
+  email: Yup.string()
+    .email('Invalid email')
+    .required('Required'),
+  password: Yup.string()
+    .required("Please enter your password")
+    .matches(
+      AUTH_VALIDATION_SETS.PASSWORD_MATCH_STRING,
+      `Password must contain at least ${AUTH_VALIDATION_SETS.PASSWORD_LENGTH} characters, one uppercase, one number and one special case character`
+    ),
+});
+
+const INITIAL_VALUES = {
+  firstName: '',
+  lastName: '',
+  email: '',
+  password: '',
+}
+
+const SignUpForm = () => {
+  const [{ fetching, error, data }, createUser] = useMutation(CREATE_USER);
+  const onSubmit = async (values, { setSubmitting }) => {
+    await createUser(values);
+    setSubmitting(false);
+  }
+
+  return (
+    <div>
+      <Formik
+        initialValues={INITIAL_VALUES}
+        validationSchema={SIGNUP_SCHEMA}
+        onSubmit={onSubmit}
+      >
+        {({ isSubmitting }) => (
+          <Grid item xs={12} sm={8} md={5}>
+            <Box
+              sx={{
+                my: 8,
+                mx: 4,
+                display: 'flex',
+                flexDirection: 'column',
+                alignItems: 'center',
+              }}
+            >
+              <Typography variant="h3" component="div" gutterBottom align="center">
+                Sign Up
+              </Typography>
+
+              {(data && data.createUser !== null) &&
```

Comment:
Use state to track this

Reply:
@VladWithCookies will use a hook useState above ?

or something like: `const success = data && data.createUser !== null;`  ?

Reply:
@AlexNGerman let's discuss this